Dockerfile

```diff
@@ -1 +1,185 @@
+ARG GO_SRC="https://golang.org/dl/go${GO_VER}.${OS}-${ARCH}.tar.gz"
+
+#
+# Set the Go environment
+#
+ENV GOROOT="/usr/local/go"
+ENV GOPATH="/go"
+ENV PATH="${PATH}:${GOROOT}/bin"
+#ENV GO_FLAGS="-tags=libpfm,netgo,libipmctl"
+ENV GO_FLAGS="-tags=libpfm,netgo"
+
+#
+# Download and install go
+#
+RUN curl -L "${GO_SRC}" -o - | tar -C "/usr/local" -xzf -
+
+#
+# Some important labels
+#
+LABEL ORG="Armedia LLC"
+LABEL MAINTAINER="Armedia Devops Team <devops@armedia.com>"
+LABEL APP="Grafana"
+LABEL VERSION="${VER}"
+
+#
+# Download and install the compilation tools
+#
+# Missing dependencies from Alpine:
+#        zfs
+#        fortify-headers
+
+#        ipmctl \
+#        libipmctl \
+RUN yum -y install epel-release
+RUN yum -y install \
+        bash \
+        cmake \
+        compat-glibc \
+        device-mapper \
+        device-mapper-persistent-data \
+        findutils \
+        gcc \
+        gcc-c++ \
+        git \
+        kernel-headers \
+        make \
+        ndctl-devel \
+        patch \
+        python3 \
+        pkgconfig \
+        wget && \
+    echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf && \
+    yum -y clean all
+
+#
+# Set the working directory for all builds
+#
+WORKDIR /usr/src/app/
+
+#
+# Build libpfm
+#
+RUN curl -L "${PFM_SRC}" -o "${PFM_TAR}" && \
+    echo "${PFM_SHA}  ${PFM_TAR}" | sha1sum -c && \
+    tar -xzf "${PFM_TAR}" && \
+    cd "libpfm-${PFM_VER}" && \
+    export DBG="-g -Wall" && \
+    make && \
+    make install
+
+#
+# Build cAdvisor
+#
+WORKDIR "${GOPATH}/src/github.com/google/cadvisor"
+COPY --from=src /src "${GOPATH}/src/github.com/google/cadvisor"
+RUN ./build/build.sh
+
+#
+# The actual runnable container
+#
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+
+#
+# Basic Parameters
+#
+ARG ARCH="amd64"
+ARG OS="linux"
+ARG VER="0.40.0"
+ARG PKG="google_cadvisor"
+ARG SRC="https://github.com/google/cadvisor.git"
+ARG UID="root"
+
+#
+# Some important labels
+#
+LABEL ORG="Armedia LLC"
+LABEL MAINTAINER="Armedia Devops Team <devops@armedia.com>"
+LABEL APP="Google cAdvisor"
+LABEL VERSION="${VER}"
+
+#
+# Some useful environment variables
+#
+ENV GOROOT="/usr/local/go"
+ENV GOPATH="/go"
+ENV CADVISOR_HEALTHCHECK_URL="http://localhost:8080/healthz"
+
+#
+# Missing dependencies from Alpine:
+#       zfs
+RUN yum -y install \
+        compat-glibc \
+        device-mapper \
+        device-mapper-persistent-data \
+        findutils \
+        libpfm \
+        ndctl \
+        wget && \
+    echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf && \
+    yum -y clean all
+
+#
+# Copy the built artifacts
+#
+COPY --from=build /usr/local/lib/libpfm.so* /usr/local/lib/
+# COPY --from=build /usr/local/lib/libipmctl.so* /usr/local/lib/
+COPY --from=build "${GOPATH}/src/github.com/google/cadvisor/cadvisor" /usr/local/bin/cadvisor
+
+#
+# Final parameters
+#
+USER        ${UID}
+EXPOSE      8080
```

Comment:
Is the port number fixed? Do we want to make it a parameter to this Docker image?

Reply:
EXPOSE is a static parameter that can't be changed at runtime ... if you want to remap it, it should be remapped at the upper level.

Reply:
I know. I am just asking whether the port is configurable or whether it is hardcoded? This question is not related to the EXPOSE command, sorry for the confusion.

Reply:
It's configurable via CLI flag

Reply:
I don't understand your answer...

Furthermore, I can see the `CADVISOR_HEALTHCHECK_URL` has port 8080 hardcoded in it, so as it stands it is not possible to change the port, whether at build time nor at run time.

Reply:
Anyway, I just wanted to understand whether the port was configurable or not, and if not whether we want to make so or not. No big deal.